In EWS, Set Credentials Before Checking Out Source

chgibb-apple opened the pull request:
#### e8d3a25226a38e7efab130a2717f1f89c84089c4
<pre>
In EWS, Set Credentials Before Checking Out Source
<a href="https://rdar.apple.com/149307815">rdar://149307815</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=291573">https://bugs.webkit.org/show_bug.cgi?id=291573</a>

Reviewed by Aakash Jain.

In EWS, Set Credentials Before Checking Out Source

* Tools/CISupport/ews-build/factories.py:
(Factory.__init__):
(StyleFactory.__init__):
(WatchListFactory.__init__):
(SaferCPPStaticAnalyzerFactory.__init__):
(CommitQueueFactory.__init__):
(MergeQueueFactoryBase.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(SetCredentialHelper):
(SetCredentialHelper.__init__):
(SetCredentialHelper.doStepIf):
(SetCredentialHelper.run):
(FetchBranches.run):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckOutPullRequest.test_success):
(TestCheckOutPullRequest.test_success_apple):
(TestCheckOutPullRequest.test_success_win):
(TestCheckOutPullRequest.test_failure):

Canonical link: <a href="https://commits.webkit.org/293894@main">https://commits.webkit.org/293894@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f82668c21b009ecd28f004ec66e2d82498209c5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99884 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19532 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9858 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105011 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50465 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101925 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19837 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27967 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76036 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33126 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102891 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15132 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90241 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56394 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14936 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8228 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49834 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84862 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8315 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107372 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26997 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19727 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84989 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/99442 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27359 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86441 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84511 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21545 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29189 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6943 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20816 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26934 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26745 "Built successfully") | | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30061 "Build is in progress. Recent messages:OS: Sequoia (15.2), Xcode: 16.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28304 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -886,6 +886,28 @@ def run(self):
+
+        return super().run()
+
+    def hideStepIf(self, results, step):
```
Any specific reason to hide this step? If not, maybe we can show it (and delete this `hideStepIf` method).

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -886,6 +886,28 @@ def run(self):
+    haltOnFailure = False
+
+    def __init__(self, **kwargs):
+        super().__init__(timeout=5 * 60, logEnviron=False, **kwargs)
```
Do you expect to need a 5 minutes timeout, or maybe 2 minutes is enough for this?

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -886,6 +886,28 @@ def run(self):
+    name = 'set-credential-helper'
+    descriptionDone = ['Set Credential Helper']
+    flunkOnFailure = False
+    haltOnFailure = False
```
Do you also want `warnOnFailure = False` ?

aj062 reviewed (commented):
Better to remove this logic from `FetchBranches` as well. Currently it will duplicate it.

Also, can you please provide any sample builds?

aj062 reviewed (commented):
Currently we have unit-tests for most of the steps in this file. We try to maintain a good unit-tests coverage for this code. Would be nice if you can also write a simple unit-test for this build-step being added. (maybe similar to `TestFetchBranches`).

briannafan commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -886,6 +886,26 @@ def run(self):
 
+class SetCredentialHelper(steps.ShellSequence, ShellMixin):
+    name = 'set-credential-helper'
+    descriptionDone = ['Set Credential Helper']
```
Nit: Capitalization change to `Set credential helper`?

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -6176,6 +6196,7 @@ def evaluateCommand(self, cmd):
                 if self.getProperty('github.number', ''):
                     self.build.addStepsAfterCurrentStep([
                         CleanGitRepo(),
+                        SetCredentialHelper(),
```
Do you actually want to re-run `SetCredentialHelper ` step if `PushCommitToWebKitRepo` fails?

JonWBedard replied:
We don't need this step here.

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -6191,6 +6212,7 @@ def evaluateCommand(self, cmd):
                 else:
                     self.build.addStepsAfterCurrentStep([
                         CleanGitRepo(),
+                        SetCredentialHelper(),
```
Or here. Since we are guaranteed to have already run it.

aj062 commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -3905,8 +3905,8 @@ def test_success_apple(self):
                 env=self.ENV,
-                command=['git', 'cherry-pick', '--allow-empty', 'HEAD..remotes/Contributor-apple/eng/pull-request-branch'],
-            ) + 0,
+                command=['git', 'cherry-pick', '--allow-empty', 'HEAD..remotes/Contributor-apple/eng/pull-request-branch']
```
These changes in `TestCheckOutPullRequest` seems completely unrelated to this PR. 
Is this a style fix for something style checker was complaining? If so, might be better to fix all similar issue in a separate PR. There seems to be many other places using similar formatting, e.g.: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps_unittest.py#L3527

aj062 reviewed (approved):
r+ (with a comment), assuming this is tested.

chgibb-apple committed:
In EWS, Set Credentials Before Checking Out Source
rdar://149307815
https://bugs.webkit.org/show_bug.cgi?id=291573

Reviewed by Aakash Jain.

In EWS, Set Credentials Before Checking Out Source

* Tools/CISupport/ews-build/factories.py:
(Factory.__init__):
(StyleFactory.__init__):
(WatchListFactory.__init__):
(SaferCPPStaticAnalyzerFactory.__init__):
(CommitQueueFactory.__init__):
(MergeQueueFactoryBase.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(SetCredentialHelper):
(SetCredentialHelper.__init__):
(SetCredentialHelper.doStepIf):
(SetCredentialHelper.run):
(FetchBranches.run):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckOutPullRequest.test_success):
(TestCheckOutPullRequest.test_success_apple):
(TestCheckOutPullRequest.test_success_win):
(TestCheckOutPullRequest.test_failure):

Canonical link: https://commits.webkit.org/293894@main
